model/src/funTest/kotlin/utils/DatabaseProvenanceFileStorageFunTest.kt

```diff
@@ -58,14 +58,14 @@ private fun TestConfiguration.writeTempFile(content: String): File =
         writeText(content)
     }
 
-class PostgresProvenanceFileStorageFunTest : WordSpec({
+class DatabaseProvenanceFileStorageFunTest : WordSpec({
```

Comment:
I believe is this renaming was done consistently, it'd be a large change.
From this diff I can see the following postgres specific things still
- PostgresListener
- PostgresStorageConfiguration
- val postgresStorage

Apart from that there are further storages, like ScanResultsStorage which also use the term postgres. Also AFAIK these only have been tested with Postgres, so generalizing might open the way for untested territory. Furthermore, do you think we may run into the situation / need to introduce postgres specific code? (It would become more difficult if we generalize). 

Given that, I'm not conviced yet that this generalization is the right way.
What do you think? Any other opinions @oss-review-toolkit/core-devs  ?

Reply:
`PostgresListener` *is* PostgreSQL-specific as it uses `EmbeddedPostgres`. Similar for `PostgresStorageConfiguration` / `PostgresConnection` as they use various PostgreSQL-specific properties (like `sslmode`).

> Also AFAIK these only have been tested with Postgres, so generalizing might open the way for untested territory.

Which is fine, IMO. People also already use `HttpFileStorage` with [untested WebDAV servers](https://github.com/oss-review-toolkit/ort/issues/6765), and I'm grateful for these reports. Documenting tested infrastructure via the class name is a bad idea anyway, IMO.

> Furthermore, do you think we may run into the situation / need to introduce postgres specific code?

We already have that, see the use of `org.postgresql.util.PGobject` in [JsonbSupport](https://github.com/oss-review-toolkit/ort/blob/main/scanner/src/main/kotlin/storages/utils/JsonbSupport.kt).

> What do you think?

I think it's a step into the right direction.

Reply:
In the server development we use [large objects](https://jdbc.postgresql.org/documentation/binary-data/) to store binary data because it is more memory efficient. This allows to stream the data directly from the database to the API endpoint without ever loading it into memory. As #7118 suggests to change the API to use streams as well, maybe we should first decide if we want to implement this with a Postgres specific feature before making the name more generic.

Reply:
Just to be clear: I do have nothing against using PostgreSQL-specific features and name classes accordingly if they do. But I find it confusing when classes claim by name to be PostgreSQL-specific if they aren't.

Reply:
Anyway, this simple name change is already causing more churn that I'm willing to handle. Let me drop the name change from this PR.